model/project_ref.go

```diff
@@ -606,6 +606,20 @@ func (p *ProjectRef) MergeWithProjectConfig(version string) (err error) {
 	return err
 }
 
+// SetGitHubAppCredentials updates or creates an entry in
+// GithubAppAuth for the project ref. If the provided values
+// are empty, the entry is deleted.
+func (p *ProjectRef) SetGithubAppCredentials(appID int64, privateKey []byte) error {
+	if appID == 0 || privateKey == nil || len(privateKey) == 0 {
```

Comment:
I think we should remove if _both_ are undefined and error and do nothing if only one is. (and adjust the tests to test that)